motr/examples/example2.c

```diff
@@ -52,29 +52,25 @@ enum {
 	KV_LEN   = 32,
 };
 
-static int index_launch_wait(struct m0_entity *e,
-			     struct m0_op    **ops,
-			     int               rc,
-			     int              *sm_rc)
+static int op_launch_wait_fini(struct m0_entity *e,
+			       struct m0_op    **ops,
+			       int              *sm_rc)
 {
-	if (rc == 0) {
-		m0_op_launch(ops, 1);
-		rc = m0_op_wait(ops[0],
-				    M0_BITS(M0_OS_FAILED,
-					    M0_OS_STABLE),
-				    M0_TIME_NEVER);
-		printf("operation rc: %i\n", ops[0]->op_rc);
-		if (sm_rc != NULL) {
-			/* Save retcodes. */
-			*sm_rc = ops[0]->op_rc;
-		}
-	} else
-		printf("operation rc: %i\n", rc);
+	int rc;
+
+	m0_op_launch(ops, 1);
+	rc = m0_op_wait(ops[0], M0_BITS(M0_OS_FAILED, M0_OS_STABLE),
+			M0_TIME_NEVER);
+	printf("rc=%d op_rc=%d\n", rc, ops[0]->op_rc);
+	if (sm_rc != NULL) {
+		/* Save retcodes. */
+		*sm_rc = ops[0]->op_rc;
+	}
 	m0_op_fini(ops[0]);
 	m0_op_free(ops[0]);
 	ops[0] = NULL;
 	m0_entity_fini(e);
```

Comment:
Suggestion: create a separate function for these 4 lines (`op_entity_fini()` or something) and use it here and below at `index_delete()`.

Reply:
Done.